### DISCUSSION TITLE
Auto-numbering and Cross-referencing Figures

### DISCUSSION THREAD

**Discussion by benilio** [facebook/docusaurus#6935] [Q&A] [answered]
I've been trying to automate the numbering of images so whenever my team and I update the documentation inserting a new image in the middle of the text we don't need to recaption all images manually.

For example: I have figures 1, 2, 3 and 4, but want to add a figure between figures 2 and 3, thus making the new image my new figure 3, the old one would turn into figure 4, and so forth.

It would also be of advantage if we could rename the references for the figures used through the text to the new ones.

The only solution I found was for R Markdown, here: https://gist.github.com/benmarwick/f3e0cafe668f3d6ff6e5#file-captions_and_crossrefs-rmd-L1

Does anyone have any idea how could I achieve that?

**Comment by slorber** [comment]
🙉 I have no idea what you are talking about

Please give a concrete example: provide a sample md file before/after the edits and show the problem with screenshots or whatever can help

**Reply by benilio** [reply]
Sorry, english is not my native or main language, so that also didn't help on my explanation. 😰

But, as @Josh-Cena pointed out the possibility for creating plugins, I could find a way to achieve my goal in the end!

Sorry for the confused ask and thanks for trying to help!

**Comment by Josh-Cena** [comment, marked as answer]
Doesn't our MDX plugin documentation have an example exactly for that? https://docusaurus.io/docs/markdown-features/plugins#creating-new-rehyperemark-plugins

Just update the selector to `'image'` instead? We never supported links to images, so you want something like that, it would probably be implemented with Remark anyways

**Reply by benilio** [reply]
That helped A LOT, I don't know why I haven't noticed this before... It wasn't as simple as just changing the selector to `image` though, but I found a another way and created a plugin that finds the figure captions I created manually and updates then to a crescent number version.

I also found a way to reference the images on text adding and `id` to then so I can put a link directly to then.

So, thanks!
